### PR TITLE
ruby: Add support for "rubocop" language server

### PR DESCRIPTION
**Pull request by vitallium**
Hi, this pull request adds support for `rubocop` language server. I noticed that `ruby-lsp` LS is becoming more popular but it still lacks diagnostics support in Zed. To cover that missing feature, it could be good to use `rubocop` LS to show diagnostics alongside with the running Ruby LSP.

Release Notes:

- N/A

### CODE REVIEW

**Comment by maxdeviant** [comment]
Is Rubocop intended to run alongside `ruby-lsp` and/or `solargraph`?

**Comment by vitallium** [comment]
> Is Rubocop intended to run alongside `ruby-lsp` and/or `solargraph`?

It's intended to run alongside `ruby-lsp`. `Solargraph` displays diagnostics if it is configured accordingly (see the `reporters` section at [this link](https://solargraph.org/guides/configuration)). However, until [this issue](https://github.com/zed-industries/zed/issues/13107) is resolved, it could be useful to have the ability to run `rubocop` LS. Additionally, there might be users who prefer to use `rubocop` LS exclusively. 

Just to add more context. [Rubocop LS](https://docs.rubocop.org/rubocop/usage/lsp.html) offers offense detection and autocorrection only.

**Comment by maxdeviant** [comment]
> > Is Rubocop intended to run alongside `ruby-lsp` and/or `solargraph`?
> 
> It's intended to run alongside `ruby-lsp`. `Solargraph` displays diagnostics if it is configured accordingly (see the `reporters` section at [this link](https://solargraph.org/guides/configuration)). It's up to the user whether LS should report diagnostics using `solargraph` or `rubocop.
> 
> However, until [this issue](https://github.com/zed-industries/zed/issues/13107) is resolved, it could be useful to have the ability to run `rubocop` LS. Additionally, there might be users who prefer to use `rubocop` LS exclusively.
> 
> Just to add more context. [Rubocop LS](https://docs.rubocop.org/rubocop/usage/lsp.html) offers offense detection and autocorrection only. I hope that answers your question.

Okay, I'm just trying to figure out how we want to represent it in the default settings.

Right now only `solargraph` runs by default and users have to opt-in to `ruby-lsp`: https://github.com/zed-industries/zed/blob/1607e4e0e14779252f3e04dee7e592ad32f841a3/assets/settings/default.json#L800-L802

So I'm wondering if we we should also add `"!rubocop"` to that list so that it doesn't run by default.

**Comment by vitallium** [comment]
> > > Is Rubocop intended to run alongside `ruby-lsp` and/or `solargraph`?
> > 
> > 
> > It's intended to run alongside `ruby-lsp`. `Solargraph` displays diagnostics if it is configured accordingly (see the `reporters` section at [this link](https://solargraph.org/guides/configuration)). It's up to the user whether LS should report diagnostics using `solargraph` or `rubocop.
> > However, until [this issue](https://github.com/zed-industries/zed/issues/13107) is resolved, it could be useful to have the ability to run `rubocop` LS. Additionally, there might be users who prefer to use `rubocop` LS exclusively.
> > Just to add more context. [Rubocop LS](https://docs.rubocop.org/rubocop/usage/lsp.html) offers offense detection and autocorrection only. I hope that answers your question.
> 
> Okay, I'm just trying to figure out how we want to represent it in the default settings.
> 
> Right now only `solargraph` runs by default and users have to opt-in to `ruby-lsp`:
> 
> https://github.com/zed-industries/zed/blob/1607e4e0e14779252f3e04dee7e592ad32f841a3/assets/settings/default.json#L800-L802
> 
> So I'm wondering if we should also add `"!rubocop"` to that list so that it doesn't run by default.

Ah, indeed. I will update settings to have `rubocop` disabled by default.

File: extensions/ruby/src/language_servers/rubocop.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+impl Rubocop {
+    pub const LANGUAGE_SERVER_ID: &'static str = "rubocop";
+
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn server_script_path(&mut self, worktree: &zed::Worktree) -> Result<String> {
+        let path = worktree.which("rubocop").ok_or_else(|| {
```
Comment (vitallium) [inline extensions/ruby/src/language_servers/rubocop.rs:13]: I just realised that all language servers in Ruby extension do not respect the `binary` settings that Zed provides to set the binary path. I will address that in a follow-up pull request.

**Comment by vitallium** [comment]
> > > > Is Rubocop intended to run alongside `ruby-lsp` and/or `solargraph`?
> > > 
> > > 
> > > It's intended to run alongside `ruby-lsp`. `Solargraph` displays diagnostics if it is configured accordingly (see the `reporters` section at [this link](https://solargraph.org/guides/configuration)). It's up to the user whether LS should report diagnostics using `solargraph` or `rubocop. However, until [this issue](https://github.com/zed-industries/zed/issues/13107) is resolved, it could be useful to have the ability to run `rubocop`LS. Additionally, there might be users who prefer to use`rubocop` LS exclusively.
> > > Just to add more context. [Rubocop LS](https://docs.rubocop.org/rubocop/usage/lsp.html) offers offense detection and autocorrection only. I hope that answers your question.
> > 
> > 
> > Okay, I'm just trying to figure out how we want to represent it in the default settings.
> > Right now only `solargraph` runs by default and users have to opt-in to `ruby-lsp`:
> > https://github.com/zed-industries/zed/blob/1607e4e0e14779252f3e04dee7e592ad32f841a3/assets/settings/default.json#L800-L802
> > 
> > So I'm wondering if we should also add `"!rubocop"` to that list so that it doesn't run by default.
> 
> Ah, indeed. I will update settings to have `rubocop` disabled by default.

Updated.

**Review by maxdeviant — APPROVED** [review]
Thank you!

**Pull request merged**